src/diffusers/models/unet_2d_blocks.py

```diff
@@ -804,7 +822,7 @@ def __init__(
         resnet_act_fn: str = "swish",
         resnet_groups: int = 32,
         resnet_pre_norm: bool = True,
-        attn_num_head_channels=1,
```

Comment:
We replaced `attn_num_head_channels` with `attn_head_dim` here: 
https://github.com/huggingface/diffusers/pull/3797/files#diff-6772eedf129163896c86d8423027ba04081212c2733ce1eca3ce98f9208b7670R741

Why are we using `attn_num_heads` here then? This create confusion for me.

Reply:
1.) `attn_num_head_channels` was a bad name. This PR gets rid of all `attn_num_head_channels`. 
2.) Here we used the wrong concept. We need to use # of heads here instead of attention head dim. That's why the change

Reply:
Cool! Thanks for explaining.